tests/e2e/default/db/initial-replication.wdio-spec.js

```diff
@@ -0,0 +1,237 @@
+  expect(localDocIds).to.include.members(translationIds);
+
+  const localForms = await browserUtils.getDocs(formIds);
+  const expectedAttachments = ['model.xml', 'form.html', 'xml'];
+  localForms.forEach(form => {
+    const attachments = form._attachments;
+    const serverForm = forms.find(serverForm => form._id === serverForm.id);
+
+    expect(Object.keys(attachments)).to.have.members(expectedAttachments, `${form._id} has incorrect attachments`);
+    expectedAttachments.forEach(attName =>
+      expect(attachments[attName].data).to.deep.equal(serverForm.doc._attachments[attName].data)
+    );
+  });
+
+  expect(localDocIds).to.include.members(ids(userAllowedDocs.clinics));
+  expect(localDocIds).to.include.members(ids(userAllowedDocs.persons));
+  expect(localDocIds).to.include.members(ids(userAllowedDocs.reports));
+
+  const replicatedDeniedDocs = _.intersection(
+    localDocIds,
+    ids([...userDeniedDocs.clinics, ...userDeniedDocs.persons, ...userDeniedDocs.reports])
+  );
+  expect(replicatedDeniedDocs).to.deep.equal([]);
+
+  const initalReplicationLog = await browserUtils.getDoc(LOCAL_LOG);
+  expect(initalReplicationLog.complete).to.equal(true);
+};
+
+const refreshAndWaitForAngular = async () => {
+  await browser.refresh();
+  await commonPage.waitForAngularLoaded(3000);
+};
+
+describe('initial-replication', () => {
+  before(async () => {
+    // we're creating ~2000 docs
+    await utils.saveDocs([...userAllowedDocs.places, ...userDeniedDocs.places]);
+    await utils.createUsers([userAllowedDocs.user]);
+
+    await sentinelUtils.waitForSentinel();
+    await utils.stopSentinel();
+
+    await utils.saveDocs(userAllowedDocs.clinics);
+    await utils.saveDocs(userDeniedDocs.clinics);
+
+    await utils.saveDocs(userAllowedDocs.persons);
+    await utils.saveDocs(userDeniedDocs.persons);
+
+    await utils.saveDocs(userAllowedDocs.reports);
+    await utils.saveDocs(userDeniedDocs.reports);
+  });
+
+  after(async () => {
+    await sentinelUtils.skipToSeq();
+    await utils.startSentinel();
+  });
+
+  afterEach(async () => {
+    await browser.reloadSession();
+    await browser.url('/');
+  });
+
+  it('should log user in', async () => {
+    await loginPage.login(userAllowedDocs.user);
+
+    await validateReplication();
+
+    // does not restart initial replication on refresh
+    await browser.refresh();
+    await commonPage.waitForAngularLoaded(3000);
+
+    // supports reloading the page while offline
+    await browser.throttle('offline');
+    await refreshAndWaitForAngular();
+    await browser.throttle('online');
+
+    // it should not restart initial replication if the local doc is missing on refresh
+    await browserUtils.deleteDoc(LOCAL_LOG);
+    await refreshAndWaitForAngular();
+
+    // it should support reloading the page while offline
+    await browser.throttle('offline');
+    await refreshAndWaitForAngular();
+  });
+
+  it('should support "disconnects"', async () => {
```

Comment:
did you consider a test which confirms that this initial replication stuff yields the same documents as the standard replication path?  a "parity test"? 

i'm not sure how easy/hard that is but i like that more holistic pattern of test for cases like this -- where we will maintain two codepaths and expect them to achieve the same result